x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/UnaryScalarFunction.java

```diff
@@ -47,18 +46,4 @@ public final Expression field() {
     public DataType dataType() {
         return field.dataType();
     }
-
-    @Override
-    public final int hashCode() {
-        return Objects.hash(field);
-    }
-
-    @Override
-    public final boolean equals(Object obj) {
-        if (obj == null || obj.getClass() != getClass()) {
-            return false;
-        }
-        UnaryScalarFunction other = (UnaryScalarFunction) obj;
-        return Objects.equals(other.field, field);
```

Comment:
Redundant, the implementation in `Function` already checks equality based on the children (all of the expressions).